@here/harp-omv-datasource/lib/OmvDecodedTileEmitter.ts

```diff
@@ -75,6 +75,7 @@ import {
 } from "@here/harp-datasource-protocol/lib/TechniqueAttr";
 // tslint:disable-next-line:max-line-length
 import { SphericalGeometrySubdivisionModifier } from "@here/harp-geometry/lib/SphericalGeometrySubdivisionModifier";
+import { ExtrusionFeature } from "@here/harp-materials/lib/MapMeshMaterials";
```

Comment:
Can we move it somewhere else to avoid the harp-material dependency?

Reply:
Thanks for raising this, i also would like to have good place.
There is no good place in current architecture, however these are propositions
1) `harp-materials` - , because this feature and constant is actually property (limitation) of material ...still, this is more or less ok
2) `harp-datasource-protocol` ... .well, it's kitchen sink already ... but currently stands as "mapview-decoder" protocol, so also good place for all definitions shared by render&decode areas
3) `harp-omv-datasource` already depends on "materials" due to transitive dependency on `harp-lines`, so this is only minor harm ...

Reply:
OTOH, if we would given, we _already_ have dependency
harp-datasource-protocol -> harp-materials

We cannot move `ExtrusionFeature` nor `MIN_BUILDING_HEIGHT`  do `harp-datasource-protocol`, because we would end with loop.

I'll settle with specific `MapMeshMaterialDefs` containing only key constants in `harp-materials`.

Reply:
I think the main question we have to answer is, what is the purpose of our packages?
1. Is it to separate decoding and rendering code or
2. to allow standalone usage of certain features like rendering of extruded geometry.

If it is 1, I would vote for `harp-datasource-protocol` b/c as you mentioned already it is the place where we put all shared code.
If it is 2, I would vote for `harp-materials` since a user that just want's to render some extruded geometry does not want to pull the whole `harp-datasource-protocol`. I guess having the `harp-materials` dependency for that usecase is also fine b/c you actually want to render at the end. Unfortunately it breaks our first usecase b/c we pull the whole `harp-materials` into the decoder bundle.

So to conclude: Since we have the `harp-material` dependency already it's fine for now to put it to `harp-materials` but going forward we should clean this up. Maybe we even need another package that contains all the shared stuff between geometry creation and materials. Also the extrusion should be moved out of OmvDecodedTile emitter into `harp-geometry`. And finally looking at the existing `harp-lines <- harp-materials` dependency: I think this should also be cleaned up b/c the material dependency is just there to have a fallback if no material is provided.

Reply:
> Unfortunately it breaks our first usecase b/c we pull the whole harp-materials into the decoder bundle.

If we're specific in imports i.e
> import { Foo} from "@here/harp-materials/lib/FooDefs"

instead of
> import { Foo } from "@here/harp-materials/"

Then we're ok and webpack (and possibly other bundlers) doesn't pick up whole library, but only needed symbols.

Reply:
Resolved issue by moving `ExtrusionFeature` definitions to separate module, still in `harp-materials`. Now it can be used both in decoder and renderer context.